### ISSUE TITLE
Recodring and other directory

### ISSUE THREAD

**Issue by NewbieProger** [holomekc/wiremock#31]
When I use recording nad other root-dir I get execption like "FileNotFound (permission denied)"

In docker i set env `--root-dir=/home/wiremock/storage` it's different from home, but when I try save mappings `/__admin/mappings/save` I'm getting 500 error

In DockerFile I see that you give permission only for one directory `RUN adduser -u 1000 -G users -h /home/wiremock -D wiremock`

**Comment by holomekc** [comment]
Hi,
Sorry for my inactivity. I will check this next weekend.
Br,
Christopher

**Comment by holomekc** [comment]
Are you using the example of docker-compose? For me it is working fine. I mean the storage directory is created by the mounted volume. I cannot do much there on the image side. For me the permissions is rwx though. You can try to add a :rw at the end of the volumes definition. E.g.:
```yaml
    volumes:
      - ./yourLocalDir:/home/wiremock/storage:rw
```

**Comment by mrvinceit** [comment]
@holomekc, apologies in advance for piggybacking on this issue, but this appears to be similar to the issue we're having. 

We are able to load mappings and files on startup, however, creating and saving new files from the UI do not end up getting created on the disk. And on restart the previously created stubs via the UI have effectively been purged, on the original stubs saved on disk are available.

Basically, we are unable to save mappings to the "mappings" folder, the configuration is similar to the example docker-compose file, and the volume is set for read-write. 

We are able to use the APIs to output the mappings in memory but it is preferable to use the UI to create and save the stubs to disk individually.

**Comment by holomekc** [comment]
Hi. @mrvinceit. No worties. Your edit message is correct. This is the only way to persist the mapping information. Maybe the naming in the ui is a bit confusing. The "normal" save button after editing a mapping is only saving the changes in the memory of wiremock. The ui reflects more or less what is provided via http api of wiremock here. 

I cannot check it right now, but the behavior you describe sounds like you copy paste a mapping with its id and save it as another one afterwards. Wiremock does not like that and results in the strange behavior you describes. The mappings are them somehow linked together. As said I am using the provided wiremock api here. Please make sure that the new mapping does not contain any ids. I think there were two actually. Not 100% sure anymore. I can check it after my vacation. Please remind me in case I forget.
Br
Chris

**Comment by mrvinceit** [comment]
Thanks for the reply, the new mappings are not happening via copy & paste workflows. To test we've tried, using a completely fresh install in both folder and list modes and once the first stub is created there is no way to deselect it. Refreshes to the page also have it as the **active** stub. 

But to your other point, we have tried removing the id portion of the "new" stub and that seems to show some success as a workaround. 

Another question, is it possible to customize the file name that gets generated, it would be much easier to know which stub is what than with the current guid file naming.

**Comment by holomekc** [comment]
Hi @mrvinceit also can try to check that after my vacavtion. I quickly checked the api. Maybe you can try to set a name:

https://wiremock.org/docs/api/#tag/Stub-Mappings/paths/~1__admin~1mappings/post

**Comment by holomekc** [comment]
Hmm. I understand. I am not sure if I want to "run away" too far from the wiremock code. That would be something to ask from the wiremock team.

**Comment by holomekc** [comment]
Sorry again for the very late response. I would still vote for asking such a feature in wiremock itself.
